src/providers/tfc/runProvider.ts

```diff
@@ -21,10 +49,127 @@ export class RunTreeDataProvider implements vscode.TreeDataProvider<vscode.TreeI
+      createIfNone: false,
+    });
+
+    if (session === undefined) {
+      return [];
+    }
+
+    try {
+      const runs = await apiClient.listRuns({
+        params: { workspace_id: workspaceId },
+      });
+
+      if (runs.data.length === 0) {
+        return [{ label: `No runs found for ${workspaceName}` }];
+      }
+
+      const items: RunTreeItem[] = [];
+      for (let index = 0; index < runs.data.length; index++) {
+        const run = runs.data[index];
+        items.push(new RunTreeItem(run.id, workspaceName, run.attributes));
+      }
+
+      // TODO: pagination
+
+      return items;
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response?.status === 401) {
+        vscode.window.showErrorMessage(`Runs not accessible for workspace ${workspaceId}: ${error}`);
+      } else if (error instanceof Error) {
+        vscode.window.showErrorMessage(error.message);
+      } else if (typeof error === 'string') {
+        vscode.window.showErrorMessage(error);
+      }
+      return [];
+    }
   }
 
   dispose() {
     //
   }
 }
+
+export class RunTreeItem extends vscode.TreeItem {
+  constructor(public id: string, public workspaceName: string, public attributes: RunAttributes) {
+    super(attributes.message, vscode.TreeItemCollapsibleState.None);
+    this.id = id;
+
+    this.workspaceName = workspaceName;
+    this.iconPath = this.getStatusIcon(attributes.status);
+    this.description = `${attributes['trigger-reason']} ${attributes['created-at']}`;
+    this.tooltip = new vscode.MarkdownString(`
+### ID: \`${id}\`
+
+${attributes.message}
+
+ - **Status**: ${attributes.status}
+ - **Trigger reason**: ${attributes['trigger-reason']}
+ - **Created at**: ${attributes['created-at']}
+ - **Source**: ${attributes.source}
+ - **Terraform** ${attributes['terraform-version']}
+`);
+  }
+
+  private getStatusIcon(status: string): vscode.ThemeIcon {
+    switch (status) {
+      // in progress
+      case 'pending':
+      case 'fetching':
+      case 'pre_plan_running':
+      case 'queuing':
+      case 'planning':
+      case 'cost_estimating':
+      case 'policy_checking':
+      case 'apply_queued':
+      case 'applying':
+      case 'post_plan_running':
+      case 'plan_queued':
+        return new vscode.ThemeIcon('sync~spin');
+
+      case 'fetching_completed':
+      case 'pre_plan_completed':
+      case 'planned':
+      case 'cost_estimated':
+      case 'policy_override':
+      case 'policy_checked':
+      case 'confirmed':
+      case 'post_plan_completed':
+      case 'planned_and_finished':
+        return new vscode.ThemeIcon('pass');
+      case 'policy_soft_failed':
+        return new vscode.ThemeIcon('warning');
+      case 'applied':
+        return new vscode.ThemeIcon('pass-filled');
+      case 'discarded':
+      case 'canceled':
+      case 'force_canceled':
+        return new vscode.ThemeIcon('discard');
+      case 'errored':
+        return new vscode.ThemeIcon('error');
```

Comment:
We can style these with color to indicate success or failure, but that can be a followup

Reply:
I'm happy to do that but I am really struggling to understand what the best way to do it with the existing VS Code APIs https://code.visualstudio.com/api/references/theme-color

What meanings/IDs should I choose for each occasion? I don't see any appropriate "success" or "fail" ones in the long list. Or did you imply we just declare our own colours? How would I go about that? Implement custom `vscode.ThemeColor` which accepts the colour in hex format or something like that?

How do we know that will render well in all light and dark themes?

I'm leaning more on leaving it for later.